ethcore/evm/src/interpreter/mod.rs

```diff
@@ -132,7 +131,7 @@ impl<Cost: CostType> vm::Vm for Interpreter<Cost> {
 			reader.position += 1;
 
 			// TODO: make compile-time removable if too much of a performance hit.
-			do_trace = do_trace && ext.trace_next_instruction(
+			let do_trace = ext.trace_next_instruction(
```

Comment:
Actually it was used before. The point is that we stop executing `trace_next_instruction` if we receive first `false`.
Removing this condition creates unnecessary overhead of invoking tracer for every executed instruction even if tracing is disabled.

Reply:
But I guess this variable is set manually?

Reply:
It's set manually only for the first iteration, then it's being changed with every iteration of the loop. So if `trace_next_instruction` returns `false` for the first time, next iteration of the loop will not invoke `trace_next_instruction` again (due to `do_trace &&` resolving to false immediately).